### PR TITLE
Improve test coverage.

### PR DESCRIPTION
**Pull request by JayNewstrom**

### CODE REVIEW

File: auth-foundation-bootstrap/src/test/java/com/okta/authfoundationbootstrap/CredentialBootstrapTest.kt [hunk 1]
```diff
@@ -103,4 +103,9 @@ class CredentialBootstrapTest {
         assertThat(credential1.await()).isSameInstanceAs(credential2.await())
         assertThat(CredentialBootstrap.credentialDataSource.listCredentials()).hasSize(1)
     }
+
+    @Test fun testOidcClient() {
```
Comment (rajdeepnanua-okta) [inline auth-foundation-bootstrap/src/test/java/com/okta/authfoundationbootstrap/CredentialBootstrapTest.kt:107]: Nit: Given what the test is doing, maybe rename it to testOidcClientInitialization?

File: auth-foundation/src/test/java/com/okta/authfoundation/jwt/JwtParserTest.kt [hunk 2]
```diff
@@ -77,4 +77,26 @@ class JwtParserTest {
...
         assertThat(jwt1.hashCode()).isEqualTo(jwt2.hashCode())
     }
+
+    @Test fun testJwtParserNonBase64Header(): Unit = runBlocking {
+        val input = "+.eyJzdWIiOiIwMHViNDF6N21nek5xcnlNdjY5NiIsIm5hbWUiOiJKYXkgTmV3c3Ryb20iLCJlbWFpbCI6ImpheW5ld3N0cm9tQGV4YW1wbGUuY29tIiwidmVyIjoxLCJpc3MiOiJodHRwczovL2V4YW1wbGUtdGVzdC5va3RhLmNvbS9vYXV0aDIvZGVmYXVsdCIsImF1ZCI6IjBvYThmdXAwbEFQWUZDNEkyNjk2IiwiaWF0IjoxNjQ0MzQ3MDY5LCJleHAiOjE2NDQzNTA2NjksImp0aSI6IklELjU1Y3hCdGRZbDhsNmFyS0lTUEJ3ZDB5T1QtOVVDVGFYYVFUWHQybGFSTHMiLCJhbXIiOlsicHdkIl0sImlkcCI6IjAwbzhmb3U3c1JhR0d3ZG40Njk2Iiwic2lkIjoiaWR4V3hrbHBfNGtTeHVDX25VMXBYRC1uQSIsInByZWZlcnJlZF91c2VybmFtZSI6ImpheW5ld3N0cm9tQGV4YW1wbGUuY29tIiwiYXV0aF90aW1lIjoxNjQ0MzQ3MDY4LCJhdF9oYXNoIjoiZ01jR1RiaEdUMUdfbGRzSG9Kc1B6USIsImRzX2hhc2giOiJEQWVMT0ZScWlmeXNiZ3NyYk9nYm9nIn0.z7LBgWT2O-DUZiOOUzr90qEgLoMiR5eHZsY1V2XPbhfOrjIv9ax9niHE7lPS5GYq02w4Cuf0DbdWjiNj96n4wTPmNU6N0x-XRluv4kved_wBBIvWNLGu_ZZZAFXaIFqmFGxPB6hIsYKvB3FmQCC0NvSXyDquadW9X7bBA7BO7VfX_jOKCkK_1MC1FZdU9n8rppu190Gk-z5dEWegHHtKy3vb12t4NR9CkA2uQgolnii8fNbie-3Z6zAdMXAZXkIcFu43Wn4TGwuzWK25IThcMNsPbLFFI4r0zo9E20IsH4gcJQiE_vFUzukzCsbppaiSAWBdSgES9K-QskWacZIWOg"
```
Comment (rajdeepnanua-okta) [inline auth-foundation/src/test/java/com/okta/authfoundation/jwt/JwtParserTest.kt:82]: Can we capture what was done to make this JWT string invalid? Like having something like:
`val valid = "valid"
val input = "+." + valid
`

File: auth-foundation/src/test/java/com/okta/authfoundation/jwt/JwtParserTest.kt [hunk 3]
```diff
@@ -77,4 +77,26 @@ class JwtParserTest {
...
+        val parser = JwtParser(Json { ignoreUnknownKeys = true }, Dispatchers.Unconfined)
+
+        val exception = assertFailsWith<IllegalArgumentException> {
+            parser.parse(input)
+        }
+        assertThat(exception).hasMessageThat().isEqualTo("Header isn't valid base64.")
+    }
+
+    @Test fun testJwtParserNonBase64Claims(): Unit = runBlocking {
+        val input = "eyJraWQiOiJGSkEwSEdOdHN1dWRhX1BsNDVKNDJrdlFxY3N1XzBDNEZnN3BiSkxYVEhZIiwiYWxnIjoiUlMyNTYifQ.+.z7LBgWT2O-DUZiOOUzr90qEgLoMiR5eHZsY1V2XPbhfOrjIv9ax9niHE7lPS5GYq02w4Cuf0DbdWjiNj96n4wTPmNU6N0x-XRluv4kved_wBBIvWNLGu_ZZZAFXaIFqmFGxPB6hIsYKvB3FmQCC0NvSXyDquadW9X7bBA7BO7VfX_jOKCkK_1MC1FZdU9n8rppu190Gk-z5dEWegHHtKy3vb12t4NR9CkA2uQgolnii8fNbie-3Z6zAdMXAZXkIcFu43Wn4TGwuzWK25IThcMNsPbLFFI4r0zo9E20IsH4gcJQiE_vFUzukzCsbppaiSAWBdSgES9K-QskWacZIWOg"
```
Comment (rajdeepnanua-okta) [inline auth-foundation/src/test/java/com/okta/authfoundation/jwt/JwtParserTest.kt:93]: Same input as above

**Review by rajdeepnanua-okta — APPROVED** [review]
Looks good. Added some nits

File: auth-foundation/src/test/java/com/okta/authfoundation/jwt/JwtParserTest.kt (hunk 2 above)
Comment (JayNewstrom) [reply auth-foundation/src/test/java/com/okta/authfoundation/jwt/JwtParserTest.kt:82]: This was mostly a remnant of some old tests. I updated most of the tests to not do this anymore (for the same reason you're bringing up here! So thanks!).

I just copy/pasted the surrounding tests to get these validated. But it definitely makes sense to move them over to the new way of doing things so it's much easier to reason about.

**Pull request merged**